index.d.ts

```diff
@@ -15,19 +15,802 @@
+
+  interface RouteAnnounce {
+    /**
+     * Message to be announced
+     */
+    message: String,
+    /**
+     * Politeness level
+     *
+     * Defaults to 'off'
+     */
+    politeness?: 'off' | 'polite' | 'assertive'
+  }
+
+  type RequireAtLeastOne<T> = {
+    [K in keyof T]-?: Required<Pick<T, K>> & Partial<Pick<T, Exclude<keyof T, K>>>
+  }[keyof T]
+
+  interface RouteHooks {
+    before?: (to: Route, from: Route) => string | Route;
+  }
+
+  type Route = {
+    /**
+     * URI path for the route
+     */
+    path: string,
+    /**
+     * Component to load when activating the route
+     */
+    component: ComponentFactory // todo: or promise returning a component instance
+    /**
+     * Transition configuration for the route
+     */
+    transition?: RequireAtLeastOne<RouteTransition> | RouteTransitionFunction,
+    /**
+     * Extra route options
+     */
+    options?: object // todo: specify which options are available,
+    /**
+     * Message to be announced when visiting the route (often used for accessibility purposes)
+     *
+     * Can be either a `String` with the message or an object that defines the message and the
+     * politeness level
+     */
+    announce?: String | RouteAnnounce
+    /**
+     * Register hooks for the route
+     */
+    hooks?: RouteHooks
+    /**
+     * Route path parameters
+     */
+    readonly params?: {
+      [key: string]: string | number
+    }
+    /**
+     * Allows for attaching custom data to a route, either hardcoded in the
+     * route definition or asigned to the route object in a before hook
+     *
+     * Will be merged with the route params and navigation data and passed as
+     * props into the route component
+     */
+    data?: {
+      [key: string]: any
+    }
+  }
+
+  type ComponentFactory = () => void
+
+  // Launch Related
+
+  type DebugLevel = 0 | 1 | 2
+  type LogTypes = 'info' | 'warn' | 'error' | 'debug'
+
+  interface WebFont {
+    /**
+    * Name of the font family
+    */
+    family: string,
+    /**
+    * Type of font (web)
+    */
+    type: 'web',
+    /**
+    * Location to the font file (i.e. `/fonts/OpenSans-Medium.ttf`)
+    */
+    file: string
+  }
+
+  interface SdfFontWithFile {
+    /**
+      * Location of the font file (i.e. `/fonts/OpenSans-Medium.ttf`)
+      */
+    file: string
+  }
+
+  interface SdfFontWithPngJson {
+    /**
+    * Location of the font map (i.e. `'/fonts/Lato-Regular.msdf.json'`)
+    */
+    json: string,
+    /**
+    * Location of the font png (i.e. `'/fonts/Lato-Regular.msdf.png'`)
+    */
+    png: string
+  }
+
+  type SdfFont = {
+    /**
+    * Name of the font family
+    */
+    family: string,
+    /**
+    * Type of font (web)
+    */
+    type: 'msdf' | 'sdf',
+  } & (SdfFontWithFile | SdfFontWithPngJson)
+
+  type Font = WebFont | SdfFont
+
+  type ShaderEffect = {
+    name: string,
+    type: RendererShaderEffect
+  }
```

Comment:
I guess this works for now - but we should think about making this easier when the new shader / effect refactor from Jeff comes in :)

Reply:
yes, this is temporary for now, and up for general refactor